R/pkg/R/serialize.R

```diff
@@ -135,9 +135,29 @@ serializeRow <- function(row) {
   rawConnectionValue(rawObj)
 }
 
-writeRaw <- function(con, batch) {
+writeRaw <- function(con, batch, ...) {
+  dataSize <- length(batch)
   writeInt(con, length(batch))
-  writeBin(batch, con, endian = "big")
+  extraArgs <- list(...)
+
+  if ("maxArgSizeBytes" %in% names(extraArgs)) {
+    if (dataSize > extraArgs$maxArgSizeBytes) {
+      stop("data size exceeds spark.r.maxArgByteSize")
+    }
+  }
+
+  LENTMAX <- if ("maxBatchSize" %in% names(extraArgs)) {
+    extraArgs$maxBatchSize
+  } else {
+    .Machine$integer.max - 1 # Slightly less than R_LEN_T_MAX macro in R
+  }
```

Comment:
shouldn't we do the dataSize check after settling on LENTMAX?
how about something like this:
```
LENTMAX = extraArgs[["maxBatchSize"]]
if (is.null(LENTMAX)) {
  LENTMAX <- .Machine$integer.max
}
if (dataSize > LENTMAX) {
  stop("data size exceeds spark.r.maxArgByteSize or integer.max")
}
```